### ISSUE TITLE
Introduce user groups

### ISSUE THREAD

**Issue by nichtich** [gbv/cocoda#319]
We likely need user groups for review. This feature also requires extensions in login-server and jskos-server but the general properties can be summarized here:

* each user can be member of any number of user groups
* membership information is not stored in mappings or annotations
* membership is not public by default, but:
    * members may choose to make their membership public for selected groups
    * some groups may require public membership
* there is one group for each identity provider (e.g. github, orcid...) and users are automatically members of these group if identifies by the corresponding identity provider
* identity providers may deliver additional membership information, for instance GitHub authenticated user can be members of GitHub organization groups and/or GitHub team groups

To summarize important design decisions:

* information about group memberships are not configured in Cocoda but they come from login-server
* information about group memberships is not share publicly

**Comment by nichtich** [comment]
See also https://github.com/gbv/cocoda/issues/319 and https://github.com/gbv/login-server/issues/36.
